### DISCUSSION TITLE
Which commands need to be removed manually in v6.0.x?

### DISCUSSION THREAD

**Discussion by Gitoffthelawn** [hluk/CopyQ#1863] [Q&A]
Thank you for the new release!  So far so good.  One question.

The v6.0.0 release notes state:

> Internal commands (like ""Show/hide main window", Pin/Unpin, Encrypt/Decrypt)
will now be automatically updated in following application releases or
whenever the language changes. The side-effect is that only icon, shortcuts,
enabled state and list order can be changed for these commands. Old internal
commands added in previous versions (5.0.0 and lower) of the app need to be
removed manually.

Can you explain this a little more?  For example, in "Commands/Global Shortcuts" I have "Pin" and "Unpin" items from v5.0.0 (or likely before that release).  Do I need to delete them?  What exactly needs to be removed manually?

Thanks for the great cross-platform tool!

**Comment by hluk** [comment]
Everything should work well for now if you don't remove any commands. Some problems will appear only if you change the app language (old command names won't change) or if in the future releases internal commands are updated (there could be some duplicates in shortcut preferences).

Here is the list of the internal commands that should be removed (in Command dialog - F6 shortcut) and have the shortcuts reassigned in Preferences (Shortcuts config tab):

- Copy next item
- Copy previous item
- Copy second item
- Create new item
- Disable clipboard storing
- Edit clipboard
- Edit first item
- Enable clipboard storing
- Paste and copy next
- Paste and copy previous
- Paste clipboard as plain text
- Paste current date and time
- Show action dialog
- Show/hide main window
- Show main window under mouse cursor
- Show the tray menu
- Take screenshot

- Pin
- Unpin

- Tag as ...
- Remove tag ...
- Add a Tag
- Remove a Tag
- Clear all tags

- Encrypt
- Decrypt
- Encrypt (needs GnuPG)
- Decrypt and Copy
- Decrypt and Paste

**Reply by Gitoffthelawn** [reply]
Thanks Lukas.

In addition to the commands I've written, I had a couple commands you didn't list, and I wanted to verify that they should *not* be deleted.  I likely created these so long ago that I forgot I wrote them:

- Open CopyQ
- Show clipboard

Separately, what do you think about adding the current shortcut key combination after the command name string (perhaps in parenthesis)?  For all the internal commands that I assigned to global shortcut key combinations, I had changed the command name string to be the command name plus the shortcut key combination.  For example: `Screenshot (Alt+Home)`

Now that the command name string cannot be edited, it's no longer possible to manually add the shortcut key combination to the command name string, so this would be a nice addition.

BTW, just curious, what was the purpose(s) of switching to internal commands?

Also, as long as names are unique, is there any harm in just keeping my existing commands that overlap with the new internal commands?

Thanks again for the great tool.  I use it all the time, and I think it's fantastic.

**Reply by hluk** [reply]
The purpose of the change is to keep the commands up to date (for example, the commands for pinning/tagging items changed in the past) and also to make the names in sync with current language settings.

It's possible to have multiple commands with the same name.

I'm curious: Why add the shortcut in the command name? The shortcuts are usually visible in menus and toolbar.

**Reply by Gitoffthelawn** [reply]
Thanks for the info.  The 2 commands I mentioned are not internal commands, correct?

> I'm curious: Why add the shortcut in the command name? The shortcuts are usually visible in menus and toolbar.

Great question.  :)  So the keyboard shortcuts are visible in the CopyQ popup menu.

**Reply by hluk** [reply]
> Thanks for the info. The 2 commands I mentioned are not internal commands, correct?

Yes, those are not internal commands.

> > I'm curious: Why add the shortcut in the command name? The shortcuts are usually visible in menus and toolbar.
> 
> Great question. :) So the keyboard shortcuts are visible in the CopyQ popup menu.

Maybe it depends on the desktop environment.
